src/lib/Components/Home/ArtistRails/ArtistRail.tsx

```diff
@@ -179,13 +176,40 @@ class ArtistRail extends React.Component<Props, State> {
     }
   }
 
+  refreshData = () => {
+    if (this.refs.rail) {
```

Comment:
Ensures component is mounted, and fixes batch issue where refresh was occurring before mount lifecycle completes. See this [SO](https://stackoverflow.com/a/40969739/1038901) answer for where I found trick.

Reply:
Might be good to add the link to that SO answer as a comment to the code?

Reply:
@damassi is that the fix for the "Can only update a mounted or mounting component" error?

Reply:
It is indeed

Reply:
Dope, yeah maybe add the link to the SO answer as a comment

Reply:
Updated